File: app/src/main/java/com/wwxiaoqi/wechat_backup/thread/RestoreThread.java
package com.wwxiaoqi.wechat_backup.thread;

import android.app.ProgressDialog;
import android.content.Context;
import com.wwxiaoqi.wechat_backup.R;
import com.wwxiaoqi.wechat_backup.utils.ShellUtils;
import java.io.File;

public class RestoreThread extends Thread {
  private File fileDir;
  private Context context;
  private ProgressDialog progressDialog;

  public RestoreThread(Context context, File fileDir) {
    this.fileDir = fileDir;
    this.context = context;
    progressDialog = new ProgressDialog(context);
    progressDialog.setMessage(context.getString(R.string.restore_prompt));
    progressDialog.setCancelable(false);
    progressDialog.show();
  }

  @Override
  public void run() {
    try {
      sleep(3000);
      new Thread(() -> {
        String[] cmd = { "su", "source " + fileDir + "/recovery.sh" };
        ShellUtils.execCmd(cmd, true);
      }).start();
      progressDialog.setMessage(context.getString(R.string.restore_success_prompt));
      sleep(3000);
      progressDialog.dismiss();
    } catch (InterruptedException e) {
      e.printStackTrace();
    }
  }

}

File: app/src/main/java/com/wwxiaoqi/wechat_backup/thread/RestoreThread.kt
package com.wwxiaoqi.wechat_backup.thread

import android.app.ProgressDialog
import android.content.Context
import com.wwxiaoqi.wechat_backup.R
import com.wwxiaoqi.wechat_backup.utils.ShellUtils
import java.io.File

class RestoreThread(private val context: Context, private val fileDir: File) : Thread() {
  private val progressDialog: ProgressDialog = ProgressDialog(context)
  override fun run() {
    try {
      sleep(3000)
      Thread {
        val cmd = arrayOf("su", "source $fileDir/recovery.sh")
        ShellUtils.execCmd(cmd, true)
      }.start()
      progressDialog.setMessage(context.getString(R.string.restore_success_prompt))
      sleep(3000)
      progressDialog.dismiss()
    } catch (e: InterruptedException) {
      e.printStackTrace()
    }
  }

  init {
    progressDialog.setMessage(context.getString(R.string.restore_prompt))
    progressDialog.setCancelable(false)
    progressDialog.show()
  }
}